frontend/viewer/src/lib/components/ui/resizable/resizable-handle.svelte

```diff
@@ -3,28 +3,94 @@
+
+  let dragging = $state(false);
+  const draggingDebounced = new Debounced(() => dragging, 300);
+
+  type OnClickHandler = ButtonProps['onclick'];
+  const leftResizerOnClick = $derived.by(() => getPaneResizerOnClick(leftPane, rightPane));
+  const rightResizerOnClick = $derived.by(() => getPaneResizerOnClick(rightPane, leftPane));
+  function getPaneResizerOnClick(pane?: ResizablePrimitive.Pane, otherPane?: ResizablePrimitive.Pane): OnClickHandler {
+    if (!pane || !otherPane) return undefined;
+    if (pane.isCollapsed()) return undefined;
+    return () => {
+      if (otherPane.isCollapsed()) {
+        otherPane.expand();
+      } else {
+        pane.collapse();
+      }
+    };
+  }
+
+  let defaultPaneProportions = $state<[number, number]>();
+
+  onMount(async () => {
+    await tick(); // one of the panes almost definitely gets mounted after us
+    if (!leftPane || !rightPane) return;
+    const totalSize = leftPane.getSize() + rightPane.getSize();
+    defaultPaneProportions = [leftPane.getSize() / totalSize, rightPane.getSize() / totalSize];
+  });
+
+  function resetPanes() {
+    if (!leftPane || !rightPane || !defaultPaneProportions) return;
+    const totalSize = leftPane.getSize() + rightPane.getSize();
+    leftPane.resize(defaultPaneProportions[0] * totalSize);
+    rightPane.resize(defaultPaneProportions[1] * totalSize);
+  }
+
+  const showResizers = $derived(leftPane && rightPane && defaultPaneProportions && !draggingDebounced.current);
 </script>
 
 <ResizablePrimitive.PaneResizer
+  onDraggingChange={(isDragging) => {
+    dragging = isDragging;
+    restProps?.onDraggingChange?.(isDragging);
+  }}
   bind:ref
   class={cn(
     'bg-border focus-visible:ring-ring relative flex w-px items-center justify-center after:absolute after:inset-y-0 after:left-1/2 after:w-1 after:-translate-x-1/2 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-offset-1 data-[direction=vertical]:h-px data-[direction=vertical]:w-full data-[direction=vertical]:after:left-0 data-[direction=vertical]:after:h-1 data-[direction=vertical]:after:w-full data-[direction=vertical]:after:-translate-y-1/2 data-[direction=vertical]:after:translate-x-0 [&[data-direction=vertical]>div]:rotate-90',
     className,
   )}
+  ondblclick={resetPanes}
   {...restProps}
 >
   {#if withHandle}
-    <div class="bg-border z-10 flex h-4 w-3 items-center justify-center rounded-sm border">
-      <Icon icon="i-mdi-drag-vertical" class="size-2.5" />
+    <div
+      class="flex gap-1 items-center justify-center z-10 min-w-3 overflow-hidden transition-all delay-100 hover:delay-300 duration-0"
+      class:hover:min-w-max={showResizers}
+    >
+      {@render paneResizer(leftResizerOnClick, 'i-mdi-chevron-left')}
+      <div class="bg-border z-10 flex h-4 w-3 items-center justify-center rounded-sm border">
+        <Icon icon="i-mdi-drag-vertical" class="size-2.5" />
+      </div>
+      {@render paneResizer(rightResizerOnClick, 'i-mdi-chevron-right')}
```

Comment:
we should probably only show the resizers based on left/right pane being not undefined, otherwise the buttons will show up, but they won't do anything.

Reply:
I now see this is done via `showResizers and the hover min-w-max class, not really sure how that works but I guess it solves this issue

Reply:
The only reliable way I could figure out to keep the handle centered was for the whole thing to always be symmetrical. If there's only 1 "resize button" (e.g. only left) I still need the right one in the dom, so things stay centered. That's why I opted to hide them via width/overflow-hidden.